src/dlls/mscoree/unixinterface.cpp

```diff
@@ -146,6 +146,37 @@ extern "C" int coreclr_create_delegate(void*, unsigned int, const char*, const c
 #endif //FEATURE_GDBJIT
 
 //
+// Preload native assembly.
+//
+// Parameters:
+//  assemblyPath            - Absolute path of the assembly to preload
+//
+// Returns:
+//  HRESULT indicating status of the operation. S_OK if the assembly was successfully preloaded
+//
+extern "C"
+int coreclr_preload_assembly(
+            const char *assemblyPath
+)
+{
+    if (assemblyPath == NULL)
+    {
+        return E_FAIL;
+    }
+
+    int length = MultiByteToWideChar(CP_UTF8, 0, assemblyPath, -1, NULL, 0);
```

Comment:
Please use StringToUnicode to make the conversion instead.

Reply:
StringToUnicode cannot be used here directly because of `new`, which can't be used before CoreCLR is initialized. I have updated pull request description with some details